[CI] RateTests testGroupingAggregate {TestCase=<0 longs>} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #10655 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/7mlzaj6quwi7y)
- [elasticsearch-periodic-platform-support #10655 / windows-2022_checkpart3_platform-support-windows](https://gradle-enterprise.elastic.co/s/cbtiw2ennrwze)
- [elasticsearch-intake #28504 / part3](https://gradle-enterprise.elastic.co/s/4tidsi5fnfq4i)
- [elasticsearch-pull-request #92627 / part-3](https://gradle-enterprise.elastic.co/s/aqqnzz5mfrkmw)
- [elasticsearch-pull-request #92620 / checkpart3_release-tests](https://gradle-enterprise.elastic.co/s/oxsqnt2ellauw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.aggregate.RateTests.testGroupingAggregate {TestCase=<0 ints>}" -Dtests.seed=BBDBD2248A050EA6 -Dtests.locale=cs-CZ -Dtests.timezone=America/Nipigon -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.expression.function.aggregate.RateTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testGroupingAggregate%20%7BTestCase%3D%3C0%20longs%3E%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (a value equal to or greater than <0.0> and a value less than or equal to <0.0>)
     but: a value less than or equal to <0.0> <2.9542261647341555E-7> was greater than <0.0>
```

**Issue Reasons:**
- [main] 5 failures in test testGroupingAggregate {TestCase=<0 longs>} (1.7% fail rate in 300 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.3% fail rate in 158 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -564,12 +564,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   method: test {csv-spec:fork.ForkBeforeStatsByWithWhere}
   issue: https://github.com/elastic/elasticsearch/issues/134817
-- class: org.elasticsearch.xpack.esql.expression.function.aggregate.RateTests
-  method: testGroupingAggregate {TestCase=<0 longs>}
-  issue: https://github.com/elastic/elasticsearch/issues/134844
-- class: org.elasticsearch.xpack.esql.expression.function.aggregate.RateTests
-  method: testGroupingAggregate {TestCase=<0 ints>}
-  issue: https://github.com/elastic/elasticsearch/issues/134845
 - class: org.elasticsearch.cluster.metadata.MetadataCreateIndexServiceTests
   method: testIndexSettingProviderPrivateSetting
   issue: https://github.com/elastic/elasticsearch/issues/134846

--- a/x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateDoubleGroupingAggregatorFunction.java
+++ b/x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateDoubleGroupingAggregatorFunction.java
@@ -193,7 +193,7 @@ public final class RateDoubleGroupingAggregatorFunction implements GroupingAggre
     private void addRawInput(int positionOffset, IntVector groups, DoubleBlock valueBlock, LongVector timestampVector) {
         if (groups.isConstant()) {
             int groupId = groups.getInt(0);
-            Buffer buffer = getBuffer(groupId, groups.getPositionCount(), timestampVector.getLong(0));
+            Buffer buffer = getBuffer(groupId, groups.getPositionCount(), timestampVector.getLong(positionOffset));
             for (int p = 0; p < groups.getPositionCount(); p++) {
                 int valuePosition = positionOffset + p;
                 if (valueBlock.isNull(valuePosition)) {
@@ -229,7 +229,7 @@ public final class RateDoubleGroupingAggregatorFunction implements GroupingAggre
         int positionCount = groups.getPositionCount();
         if (groups.isConstant()) {
             int groupId = groups.getInt(0);
-            Buffer buffer = getBuffer(groupId, positionCount, timestampVector.getLong(0));
+            Buffer buffer = getBuffer(groupId, positionCount, timestampVector.getLong(positionOffset));
             for (int p = 0; p < positionCount; p++) {
                 int valuePosition = positionOffset + p;
                 buffer.appendWithoutResize(timestampVector.getLong(valuePosition), valueVector.getDouble(valuePosition));

--- a/x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateIntGroupingAggregatorFunction.java
+++ b/x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateIntGroupingAggregatorFunction.java
@@ -193,7 +193,7 @@ public final class RateIntGroupingAggregatorFunction implements GroupingAggregat
     private void addRawInput(int positionOffset, IntVector groups, IntBlock valueBlock, LongVector timestampVector) {
         if (groups.isConstant()) {
             int groupId = groups.getInt(0);
-            Buffer buffer = getBuffer(groupId, groups.getPositionCount(), timestampVector.getLong(0));
+            Buffer buffer = getBuffer(groupId, groups.getPositionCount(), timestampVector.getLong(positionOffset));
             for (int p = 0; p < groups.getPositionCount(); p++) {
                 int valuePosition = positionOffset + p;
                 if (valueBlock.isNull(valuePosition)) {
@@ -229,7 +229,7 @@ public final class RateIntGroupingAggregatorFunction implements GroupingAggregat
         int positionCount = groups.getPositionCount();
         if (groups.isConstant()) {
             int groupId = groups.getInt(0);
-            Buffer buffer = getBuffer(groupId, positionCount, timestampVector.getLong(0));
+            Buffer buffer = getBuffer(groupId, positionCount, timestampVector.getLong(positionOffset));
             for (int p = 0; p < positionCount; p++) {
                 int valuePosition = positionOffset + p;
                 buffer.appendWithoutResize(timestampVector.getLong(valuePosition), valueVector.getInt(valuePosition));

--- a/x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateLongGroupingAggregatorFunction.java
+++ b/x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateLongGroupingAggregatorFunction.java
@@ -193,7 +193,7 @@ public final class RateLongGroupingAggregatorFunction implements GroupingAggrega
     private void addRawInput(int positionOffset, IntVector groups, LongBlock valueBlock, LongVector timestampVector) {
         if (groups.isConstant()) {
             int groupId = groups.getInt(0);
-            Buffer buffer = getBuffer(groupId, groups.getPositionCount(), timestampVector.getLong(0));
+            Buffer buffer = getBuffer(groupId, groups.getPositionCount(), timestampVector.getLong(positionOffset));
             for (int p = 0; p < groups.getPositionCount(); p++) {
                 int valuePosition = positionOffset + p;
                 if (valueBlock.isNull(valuePosition)) {
@@ -229,7 +229,7 @@ public final class RateLongGroupingAggregatorFunction implements GroupingAggrega
         int positionCount = groups.getPositionCount();
         if (groups.isConstant()) {
             int groupId = groups.getInt(0);
-            Buffer buffer = getBuffer(groupId, positionCount, timestampVector.getLong(0));
+            Buffer buffer = getBuffer(groupId, positionCount, timestampVector.getLong(positionOffset));
             for (int p = 0; p < positionCount; p++) {
                 int valuePosition = positionOffset + p;
                 buffer.appendWithoutResize(timestampVector.getLong(valuePosition), valueVector.getLong(valuePosition));

--- a/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/X-RateGroupingAggregatorFunction.java.st
+++ b/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/X-RateGroupingAggregatorFunction.java.st
@@ -193,7 +193,7 @@ public final class Rate$Type$GroupingAggregatorFunction implements GroupingAggre
     private void addRawInput(int positionOffset, IntVector groups, $Type$Block valueBlock, LongVector timestampVector) {
         if (groups.isConstant()) {
             int groupId = groups.getInt(0);
-            Buffer buffer = getBuffer(groupId, groups.getPositionCount(), timestampVector.getLong(0));
+            Buffer buffer = getBuffer(groupId, groups.getPositionCount(), timestampVector.getLong(positionOffset));
             for (int p = 0; p < groups.getPositionCount(); p++) {
                 int valuePosition = positionOffset + p;
                 if (valueBlock.isNull(valuePosition)) {
@@ -229,7 +229,7 @@ public final class Rate$Type$GroupingAggregatorFunction implements GroupingAggre
         int positionCount = groups.getPositionCount();
         if (groups.isConstant()) {
             int groupId = groups.getInt(0);
-            Buffer buffer = getBuffer(groupId, positionCount, timestampVector.getLong(0));
+            Buffer buffer = getBuffer(groupId, positionCount, timestampVector.getLong(positionOffset));
             for (int p = 0; p < positionCount; p++) {
                 int valuePosition = positionOffset + p;
                 buffer.appendWithoutResize(timestampVector.getLong(valuePosition), valueVector.get$Type$(valuePosition));

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/RateTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/RateTests.java
@@ -87,7 +87,7 @@ public class RateTests extends AbstractAggregationTestCase {
             TestCaseSupplier.TypedData fieldTypedData = fieldSupplier.get();
             List<Object> dataRows = fieldTypedData.multiRowData();
             if (randomBoolean()) {
-                List<Object> withNulls = new ArrayList<>(dataRows);
+                List<Object> withNulls = new ArrayList<>(dataRows.size());
                 for (Object dataRow : dataRows) {
                     if (randomBoolean()) {
                         withNulls.add(null);

